integrations/azure-devops/azure_devops/client/azure_devops_client.py

```diff
@@ -286,24 +286,27 @@ async def get_columns(self) -> AsyncGenerator[list[dict[str, Any]], None]:
                 ]
 
     async def _enrich_boards(
-        self, boards: list[dict[str, Any]], project_id: str
+        self, boards: list[dict[str, Any]], project_id: str, team_id: str
     ) -> list[dict[str, Any]]:
         for board in boards:
             response = await self.send_request(
                 "GET",
-                f"{self._organization_base_url}/{project_id}/{API_URL_PREFIX}/work/boards/{board['id']}",
+                f"{self._organization_base_url}/{project_id}/{team_id}/{API_URL_PREFIX}/work/boards/{board['id']}",
             )
             board.update(response.json())
         return boards
 
-    async def _get_boards(self, project_id: str) -> list[dict[str, Any]]:
-        get_boards_url = (
-            f"{self._organization_base_url}/{project_id}/{API_URL_PREFIX}/work/boards"
-        )
-        response = await self.send_request("GET", get_boards_url)
-        board_data = response.json().get("value", [])
-        logger.info(f"Found {len(board_data)} boards for project {project_id}")
-        return await self._enrich_boards(board_data, project_id)
+    async def _get_boards(
+        self, project_id: str
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        teams_url = f"{self._organization_base_url}/{API_URL_PREFIX}/projects/{project_id}/teams"
+        async for teams_in_project in self._get_paginated_by_top_and_skip(teams_url):
```

Comment:
we have a generate_teams() method already. why don't you use that one?

Reply:
Generate teams gets the teams outside of the project context and it might not be suitable for this purpose

Reply:
I see